jgrapht-core/src/main/java/org/jgrapht/alg/PathValidator.java

```diff
@@ -27,8 +27,9 @@
  * 
  * @author Assaf Mizrachi
  * @since July, 21, 2016
- *
+ * @deprecated Use {@link org.jgrapht.alg.shortestpath.PathValidator} instead.
```

Comment:
Assuming we are going to resolve https://github.com/jgrapht/jgrapht/pull/323, shouldn't this be moving to alg.interfaces instead of alg.shortestpath?  It's similar to AStarAdmissibleHeuristic (an interface which the caller of the algorithm implements in order to influence the algorithm's behavior).